### ISSUE TITLE
Release?

### ISSUE THREAD

**Issue by bwiernik** [citation-style-language/styles#6121]
Are we expecting automatic release tags with each commit?

https://github.com/citation-style-language/styles/releases/tag/v0.0.1

@retorquere @adam3smith

**Comment by retorquere** [comment]
No, just a publish when merge to main happens.

**Comment by adam3smith** [comment]
Isn't this part of the new Composer integration?

**Comment by retorquere** [comment]
Wait, I'd missed that. So every merge to main would get a new tag bump? Why do we want that?

**Comment by adam3smith** [comment]
As I understand it, composer needs some type of versioning and since we're versioning in branches, this won't hurt (otoh, I also would not mind something less visible)

**Comment by retorquere** [comment]
It's more that for my projects, a push to master is never automatically a new release. A release is a deliberate step.

**Comment by retorquere** [comment]
I don't know, honestly. If that's what works for the style repo I'm good. I would usually want to have an answer to

> what is a new release in the context of the style repository?

before implementing it.

**Comment by bwiernik** [comment]
I would want to avoid the GitHub release system if possible. That's a lot of tags, and I don't see the point in numbering them. It's also really distracting in the notification feed.

**Comment by retorquere** [comment]
The GH release system is separate from the tags though.

**Comment by adam3smith** [comment]
I don't know enough about composer to say what the minimal thing we can do to make this happen. I'd hope that tagging but not releasing would be sufficient, but not sure. 
Looping in @seboettg  and @hxdef 
(FWIW, the original request to do this dates 5 years back and the merged PR was under review for 1 year)

**Comment by seboettg** [comment]
Hi @adam3smith 
Thank you for taking care of this! Composer/Packagist needs tags on Github in order to point a specific snapshot of the repo.

I added an entry in packagist for the styles repo and I was able to include the styles repository as an requirement into citeproc-php. That works fine so far 👍 

```
...
    "require": {
        "citation-style-language/styles": "0.0.*",
        ...
    }
...
```

Beside that we need to set a github hook in order to enable packagist to recognize changes in the repo.

<img width="776" alt="grafik" src="https://user-images.githubusercontent.com/5129682/184868582-12c6b628-274b-46aa-9e9f-30e07e09d4d0.png">

On https://packagist.org/about#how-to-update-packages I found the following information:

> Enabling the Packagist service hook ensures that your package will always be updated instantly when you push to GitHub.
>
> Do not want to log in via GitHub and grant us webhook configuration access?
>
> You can configure a GitHub webhook manually by using the following values:
>
>    Payload URL: https://packagist.org/api/github?username=seboettg
>    Content Type: application/json
>    Secret: your [Packagist API Token](https://packagist.org/profile/)
>    Which events? Just the push event is enough.

I could email you the packagist token from my packagist account and you would then have to set the github hook accordingly.

**Comment by seboettg** [comment]
By the way citeproc-php needs also a composer.json file in the locale repository :)

**Comment by adam3smith** [comment]
@seboettg -- I have given packagist Webhook access for CSL and added you to the CSL organization (without any added permissions) -- could you see if that is enough for you to enable updating automatically via webhook? If it is, great, if not, let's go the manual way.

**Comment by bkosborne** [comment]
I'm also interested in this, but it seems like tags already _are_ being created?

<img width="372" alt="Screen Shot 2023-01-03 at 9 28 13 AM" src="https://user-images.githubusercontent.com/544886/210376978-5dd7f2c3-8aa2-4ff6-a66a-4fc1f2edbc7d.png">

**Comment by adam3smith** [comment]
Yeah, I think the last remaining action here is on the composer end @seboettg -- have you had a chance to look at this?

**Comment by bkosborne** [comment]
Just tested this with composer and it works as expected. This can be closed as fixed.

`composer require citation-style-language/styles:0.0.149` downloads the latest tag. The other tag versions work as well.

**Comment by adam3smith** [comment]
https://packagist.org/packages/citation-style-language/styles does say it's not auto-updating, though, which is confusing.
